### PR TITLE
Add Filing Separate but Cohabitating to NJ Property Tax Deduction/Credit

### PR DESCRIPTION
**Pull request by leogoldman**
Adjust deduction/credit/thresholds if filing separately but maintaining the same home.
Add eligiblity for property tax credit (regardless of filing threshold) if senior, blind, or disabled.

### Summary
🏠🧮🧑‍🦯

This pull request updates and improves the policy simulation for New Jersey property tax deduction and credit. It renames and refactors some variables and files, adds new variables and test cases, and fixes some bugs and edge cases. It affects the files `policyengine_us/variables/gov/states/nj/tax/income/property_tax/*.py` and `policyengine_us/tests/policy/baseline/gov/states/nj/tax/income/property_tax/*.yaml`.

> _`nj_property_tax`_
> _Credit and deduction rules_
> _Autumn leaves change code_

### Walkthrough
*  Rename `nj_property_tax_deduction_or_credit_eligible` to `nj_property_tax_deduction_eligible` and create a new variable `nj_property_tax_credit_eligible` to separate the eligibility criteria for the deduction and the credit ([link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-510c266f9126b2e021a38d8fa7555cc031b653b107ed3ca0474fd7e4a498d7c8L15-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-510c266f9126b2e021a38d8fa7555cc031b653b107ed3ca0474fd7e4a498d7c8L37-R37), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-510c266f9126b2e021a38d8fa7555cc031b653b107ed3ca0474fd7e4a498d7c8L52-R52), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-510c266f9126b2e021a38d8fa7555cc031b653b107ed3ca0474fd7e4a498d7c8L63-R63), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-510c266f9126b2e021a38d8fa7555cc031b653b107ed3ca0474fd7e4a498d7c8L74-R74), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-f0b9b29eb18a5e5468240a735a3b4f32b07eac79c6df043bd01582c2b01dfd79L5-R5), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-f0b9b29eb18a5e5468240a735a3b4f32b07eac79c6df043bd01582c2b01dfd79L15-R31), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-dde8b065c1396dc95b9944e3511df2bc66ff7444890bd30c675d3312cb49e06cL11-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-dde8b065c1396dc95b9944e3511df2bc66ff7444890bd30c675d3312cb49e06cL23-R23), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-dde8b065c1396dc95b9944e3511df2bc66ff7444890bd30c675d3312cb49e06cL35-R35), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-dde8b065c1396dc95b9944e3511df2bc66ff7444890bd30c675d3312cb49e06cL47-R47), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-a175043d58f702111d103d0b3a629fcc02d9a5cb4fb08e78ce542b9267dfc206L1-R5), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-a175043d58f702111d103d0b3a629fcc02d9a5cb4fb08e78ce542b9267dfc206L17-R17), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-a175043d58f702111d103d0b3a629fcc02d9a5cb4fb08e78ce542b9267dfc206L29-R29), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-6426b8e27d5534a7e1b566480b06923d2a14e13129814630393ce64998b0144cL11-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-b3016e28dd41de13180245721363f59d6c7de07b6513c9003edc5aa6c54070b2L11-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-1e94ef3a90b242665a859f7bab292745d84267fb8aa108c24ace4d9a63b7007dL4-R7), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-1e94ef3a90b242665a859f7bab292745d84267fb8aa108c24ace4d9a63b7007dL13-L14), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-7e067c350f86740de9c03e1f47a542c7c047dcc9344d82e1cd1e0aefe521735fL11-R14))
*  Add logic to account for the scenario where the tax unit is filing separately but maintaining the same home, which affects the property tax amount, the deduction threshold, and the credit amount ([link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-6426b8e27d5534a7e1b566480b06923d2a14e13129814630393ce64998b0144cL32-R39), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-b3016e28dd41de13180245721363f59d6c7de07b6513c9003edc5aa6c54070b2L24-R28), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-7e067c350f86740de9c03e1f47a542c7c047dcc9344d82e1cd1e0aefe521735fL70-R76))
*  Replace `np.where` with `where` in the formula for `nj_potential_property_tax_deduction` to use the OpenFisca syntax for conditional expressions ([link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-6426b8e27d5534a7e1b566480b06923d2a14e13129814630393ce64998b0144cL26-R24))
*  Remove comments that are no longer relevant from `nj_property_tax_deduction` and `nj_property_tax_deduction_eligible` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-1562d1236484512decbbc68d720a48e9c615973580f3cb5681c58fa47223a8cfL15-L16), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-1e94ef3a90b242665a859f7bab292745d84267fb8aa108c24ace4d9a63b7007dL13-L14))
*  Add a new variable `cohabitating_spouses` to indicate whether spouses in joint or separate tax units are cohabitating, which affects the property tax deduction and credit rules ([link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-c54807fd33379126b956ebc49da6d13203b892e36e3f4059e4fa21f9b525bea7R1-R11))
*  Add test cases for the new variables `nj_property_tax_credit_eligible` and `cohabitating_spouses` and update test cases for the renamed and modified variables ([link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-73299aac3e30ededbcd8efac42c2ba881013deede5efd3d3e68c4be998f27f87R1-R59), [link](https://github.com/PolicyEngine/policyengine-us/pull/2299/files?diff=unified&w=0#diff-882973cf9b148ac2e5e8a882c3eac07afa22dd20af9fbe2f83b1763d6441f804R1-R34))

### CODE REVIEW

File: policyengine_us/variables/gov/states/nj/tax/income/property_tax/nj_property_tax_credit_eligible.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            blind_head
+            + disabled_head
+            + blind_spouse
+            + disabled_spouse
+            + senior_head
+            + senior_spouse
+        ) > 0
+
+        # Next check if they paid property taxes (either directly or through rent).
+        direct_property_taxes = tax_unit("nj_homeowners_property_tax", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nj/tax/income/property_tax/nj_property_tax_credit_eligible.py:35]: Why use `nj_homeowners_property_tax` over `real_estate_taxes`?

Comment (leogoldman) [reply policyengine_us/variables/gov/states/nj/tax/income/property_tax/nj_property_tax_credit_eligible.py:35]: Thought the rules around eligible NJ property taxes (only on primary home, restrictions on rooms/bathrooms, etc.) might be more restrictive, but if it's safe to assume they're the same I can remove the NJ variable.

File: policyengine_us/variables/household/demographic/tax_unit/senior_head.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+class senior_head(Variable):
+    value_type = bool
+    entity = TaxUnit
+    definition_period = YEAR
+    label = "Tax unit head is a senior"
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        senior = person("is_senior", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/household/demographic/tax_unit/senior_head.py:12]: This is a somewhat arbitrary categorization for the purposes of poverty age breakdowns.

If the law defines age 65 for eligibility, please add a parameter specifying age 65.

File: policyengine_us/variables/gov/states/nj/tax/income/property_tax/nj_property_tax_credit_eligible.py (hunk 1 above)
Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/nj/tax/income/property_tax/nj_property_tax_credit_eligible.py:35]: Just took a closer look. I think a separate variable makes sense if it's limited to the property taxes that would count for this credit (not just all homeowner property taxes), and if you add `adds = ["real_estate_taxes"]` so that flows through by default.
